lib/spack/spack/solver/asp.py

```diff
@@ -1691,15 +1697,8 @@ def variant_value(self, pkg, name, value):
             )
             return
 
-        pkg_class = spack.repo.path.get_pkg_class(pkg)
-
-        variant = self._specs[pkg].variants.get(name)
-        if variant:
-            # it's multi-valued
-            variant.append(value)
-        else:
-            variant = pkg_class.variants[name].make_variant(value)
-            self._specs[pkg].variants[name] = variant
+        self._specs[pkg].add_variant_values(name,
+                                            value)
```

Comment:
80 chars per line is limiting, 50 is too much :laughing: 
```suggestion
        self._specs[pkg].add_variant_values(name, value)
```